packages/@aws-cdk/aws-vpcv2-alpha/lib/subnet-v2.ts

```diff
@@ -0,0 +1,399 @@
+import { Resource, Names } from 'aws-cdk-lib';
+import { CfnRouteTable, CfnSubnet, CfnSubnetRouteTableAssociation, INetworkAcl, IRouteTable, ISubnet, NetworkAcl, SubnetNetworkAclAssociation, SubnetType } from 'aws-cdk-lib/aws-ec2';
+import { Construct, DependencyGroup, IDependable } from 'constructs';
+import { IVpcV2 } from './vpc-v2-base';
+import { CidrBlock, CidrBlockIpv6 } from './util';
+
+/**
+ * Interface to define subnet CIDR
+ */
+interface ICidr {
+  readonly cidr: string;
+}
+
+/**
+ * IPv4 or IPv6 CIDR range for the subnet
+ */
+export class IpCidr implements ICidr {
+
+  /**
+ * IPv6 CIDR range for the subnet
+ * Allowed only if IPv6 is enabled on VPc
+ */
+  public readonly cidr: string;
+  constructor(props: string ) {
+    this.cidr = props;
+  }
+}
+
+/**
+ * Properties to define subnet for VPC.
+ */
+export interface SubnetV2Props {
+/**
+ * VPC Prop
+ */
+  readonly vpc: IVpcV2;
+
+  /**
+   * ipv4 cidr to assign to this subnet.
+   * See https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-subnet.html#cfn-ec2-subnet-cidrblock
+   */
+  readonly cidrBlock: IpCidr;
+
+  /**
+   * Ipv6 CIDR Range for subnet
+   * @default No Ipv6 address
+   */
+  readonly ipv6CidrBlock?: IpCidr;
```

Comment:
sorry could you remind me why we made this as Interface, and not a normal string

Reply:
yes, we wanted to keep this to include ipam feature for subnet IP allocation as well in this interface